tremor-script/src/ast/raw.rs

```diff
@@ -2392,6 +2404,28 @@ impl<'script> Upable<'script> for InvokeRaw<'script> {
                         args,
                     })
                 } else {
+                    // Relative locability
```

Comment:
This logic seems problematic as it could creat conflicting references if I understand correctly. For example if the same function is defined locally and absolute:

```tremor
mod b with
    function badger() with
      "I'm evil but possible"
    end
end;

mod a with
  mod b with
    function badger() with
      "badger"
    end;
  end;
  define fn snot() with
    badger
  end; 
end;

a::snot() # this when I read correctly would return "I'm evil but possible"
```

Reply:
A correct script:

```tremor
# FIle: script.tremor
mod b with
    fn badger() with
      "I'm evil but possible"
    end
end;

mod a with
  mod b with
    fn badger() with
      "badger"
    end;
  end;
  fn snot() with
    badger
  end; 
end;

a::snot() 
```

Correctly evaluates to the error:

```
Error in script.tremor:14:5 
   14 |     badger
      |     ^^^^^^ Trying to access a non existing local key `badger`

```

This should exhibit the same behaviour with a troy embedded query with the above as a script operator...

Reply:
ah yes, there was `()` missing in the original it would be

```
# FIle: script.tremor
mod b with
    fn badger() with
      "I'm evil but possible"
    end
end;

mod a with
  mod b with
    fn badger() with
      "badger"
    end;
  end;
  fn snot() with
    badger()
  end; 
end;
```

Reply:
Not 100% certain without tests - but I think the args/with and subquery refactoring fixes this. Questions remain on the correct hygienic use of `use` though.